lib/services/vendor/infra/vendor-service.ts

```diff
@@ -0,0 +1,208 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+import {
+  Stack,
+  aws_eks as eks,
+  aws_iam as iam,
+  aws_ec2 as ec2,
+  aws_ecr_assets as ecr_assets,
+  ArnFormat,
+} from "aws-cdk-lib";
+import { EventBus } from "aws-cdk-lib/aws-events";
+import { Construct } from "constructs";
+import * as path from "path";
+import { KubectlV27Layer } from "@aws-cdk/lambda-layer-kubectl-v27";
+import { EventbridgeToSqs } from "@aws-solutions-constructs/aws-eventbridge-sqs";
+import { SettlementEvents } from "../../settlement/infra/settlement-service";
+import { GraphWidget } from "aws-cdk-lib/aws-cloudwatch";
+import { createGraphWidget, createMetric } from "../../../observability/cw-dashboard/infra/ClaimsProcessingCWDashboard";
+
+export enum VendorEvents {
+  SOURCE = "vendor.service",
+  VENDOR_FINALIZED = "Vendor.Finalized",
+}
+
+interface VendorServiceProps {
+  readonly bus: EventBus;
+}
+
+// Attach policies that are required for read access from AWS Console
+function attachConsoleReadOnlyPolicies(scope: Construct, role: iam.Role) {
+  role.addToPolicy(
+    new iam.PolicyStatement({
+      actions: [
+        "eks:ListFargateProfiles",
+        "eks:DescribeNodegroup",
+        "eks:ListNodegroups",
+        "eks:ListUpdates",
+        "eks:AccessKubernetesApi",
+        "eks:ListAddons",
+        "eks:DescribeCluster",
+        "eks:DescribeAddonVersions",
+        "eks:ListClusters",
+        "eks:ListIdentityProviderConfigs",
+        "iam:ListRoles",
+      ],
+      resources: ["*"],
+    })
+  );
+  role.addToPolicy(
+    new iam.PolicyStatement({
+      actions: ["ssm:GetParameter"],
+      resources: [
+        Stack.of(scope).formatArn({
+          service: "ssm",
+          resource: "parameter",
+          arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+          resourceName: "*",
+        }),
+      ],
+    })
+  );
+}
+
+export class VendorService extends Construct {
+  public readonly cluster: eks.Cluster;
+  public readonly vendorMetricsWidget: GraphWidget;
+
+  constructor(scope: Construct, id: string, props: VendorServiceProps) {
+    super(scope, id);
+
+    const region = Stack.of(this).region;
+
+    const clusterAdmin = new iam.Role(this, "AdminRole", {
+      assumedBy: new iam.AccountRootPrincipal(),
```

Comment:
You may need to use a dedicated IAM role here for deployment.

Reply:
@jihed Sorry, I did not understand. Isn't `clusterAdmin` a dedicated IAM role? What am I missing here?